hard/ultrascale_plus/bsg_link/tcl/bsg_link_ddr.sample_constraints.tcl

```diff
@@ -0,0 +1,40 @@
+# User should re-define periods / names / ports / pins / margins
+# Then add the following lines into the .xdc constraint file
+
+# create input clock
+set fmc_input_clk_period         2.000
+set fmc_input_clk_name           fmc_clk_in
+set fmc_input_clk_port           [get_ports fmc_clk_i]
+create_clock -name $fmc_input_clk_name -period $fmc_input_clk_period $fmc_input_clk_port
+
+# create output clock
+set fmc_output_clk_name          fmc_clk_out
+set fmc_output_clk_pin           [get_pins uplink/ch[0].oddr_phy/ODDRE1_clk/C]
+set fmc_output_clk_port          [get_ports fmc_clk_o]
+create_generated_clock -name $fmc_output_clk_name -source $fmc_output_clk_pin -edges {1 2 3} -edge_shift {0 0 0} $fmc_output_clk_port
+
+# input delay margins
+set dv_bre                 1.0
+set dv_are                 1.0
+set dv_bfe                 1.0
+set dv_afe                 1.0
+
+# input delay constraints
+set fmc_input_data_port          [get_ports {fmc_data_i[*] fmc_v_i}]
+set_input_delay -clock $fmc_input_clk_name -max [expr $fmc_input_clk_period/2 - $dv_bre] $fmc_input_data_port
+set_input_delay -clock $fmc_input_clk_name -min $dv_are                                  $fmc_input_data_port
+set_input_delay -clock $fmc_input_clk_name -max [expr $fmc_input_clk_period/2 - $dv_bfe] $fmc_input_data_port -clock_fall -add_delay
+set_input_delay -clock $fmc_input_clk_name -min $dv_afe                                  $fmc_input_data_port -clock_fall -add_delay
+
+# output delay margins
+set tsu_r                  0.8
+set thd_r                  0.8
+set tsu_f                  0.8
+set thd_f                  0.8
+
+# output delay constraints
+set fmc_output_data_port         [get_ports {fmc_data_o[*] fmc_v_o}]
```

Comment:
Do we want to put FMC clocks in a logically exclusive clock group so that they don't time to core clock?

Reply:
We need them to be in the same clock group, so that core->output_clk path is constrained

Reply:
Right sorry I meant the FMC input clocks, those are not in core clock group correct?

CG0: [FMC in]
CG1: [core_clock, FMC_out]

Reply:
Yes you are correct. Each FMC_in clock should be in their own async group. FMC_out_io and FMC_out should be in the same group. core_clock and {FMC_out_io, FMC_out} could be set either way (you can even share core_clock with FPC_out_io)